src/components/Proposal/Create/SchemeForms/CreateGenericMultiCallProposal.tsx

```diff
@@ -0,0 +1,502 @@
+    const values = [];
+
+    for (const contract of formValues.contracts) {
+      contractsToCall.push(contract.address);
+      callsData.push(contract.callData);
+      values.push(contract.value);
+    }
+
+    const proposalValues = {
+      title: formValues.title,
+      description: formValues.description,
+      contractsToCall: contractsToCall,
+      callsData: callsData,
+      values: values,
+      url: formValues.url,
+      dao: this.props.daoAvatarAddress,
+      scheme: this.props.scheme.address,
+      tags: this.state.tags,
+    };
+
+    setSubmitting(false);
+    await this.props.createProposal(proposalValues);
+
+    Analytics.track("Submit Proposal", {
+      "DAO Address": this.props.daoAvatarAddress,
+      "Proposal Title": formValues.title,
+      "Scheme Address": this.props.scheme.address,
+      "Scheme Name": this.props.scheme.name,
+    });
+
+    this.props.handleClose();
+  }
+
+  // Exports data from form to a shareable url.
+  public exportFormValues(values: IFormValues) {
+    exportUrl({ ...values, ...this.state });
+    this.props.showNotification(NotificationStatus.Success, "Exportable url is now in clipboard :)");
+  }
+
+  private onTagsChange = (tags: any[]): void => {
+    this.setState({ tags });
+  }
+
+  /**
+ * Given a contract address, checks whether it's valid, not exists in the current contract list and that the contract is verified with valid ABI data and write methods.
+ * If all checks are okay, pushes the contract address to the contract lists, otherwise returns an appropriate message.
+ * @param {string} contractToCall
+ */
+  private verifyContract = async (contractToCall: string) => {
+    const addContractStatus: IAddContractStatus = {
+      error: "",
+      message: "",
+    };
+
+    if (contractToCall === "") {
+      addContractStatus.error = "";
+      addContractStatus.message = "";
+    } else if (!isAddress(contractToCall)) {
+      addContractStatus.error = "NOT_VALID_ADDRESS";
+      addContractStatus.message = "Please enter a valid address";
+    } else if (this.state.whitelistedContracts.includes(contractToCall) || this.state.userContracts.includes(contractToCall)) {
+      addContractStatus.error = "CONTRACT_EXIST";
+      addContractStatus.message = "Contract already exist!";
+    } else {
+      this.setState({ loading: true });
+      const abiData = await getABIByContract(contractToCall);
+      const abiMethods = extractABIMethods(abiData);
+      if (abiMethods.length > 0) {
+        this.state.userContracts.push(contractToCall);
+        addContractStatus.error = "";
+        addContractStatus.message = "Contract added successfully!";
+      } else {
+        addContractStatus.error = "ABI_DATA_ERROR";
+        addContractStatus.message = abiData.length === 0 ? "No ABI found for target contract, please verify the " : "No write methods found for target ";
+      }
+    }
+    this.setState({ loading: false, addContractStatus: addContractStatus });
```

Comment:
so the state is set even there is an error?

Reply:
Yes, the error state is used to display what kind of error occured (Not valid address, contract exist already in the list or contract has not valid ABI data)